### PR TITLE
chainlocks: Rely more on clsigs with known blocks instead of pure clsigs

### PR DESCRIPTION
**Pull request by UdjinM6**
A malicious quorum can sign a clsig with some (random) hash and some future height, up to `tip + SIGN_HEIGHT_OFFSET` (anything above it won't pass `VerifyRecoveredSig`/`SelectQuorumForSigning`, see tests for `fake_clsig2`). Such a clsig not only does not lock the chain but it also prevents new clsigs (for heights up to that future one) from forming or being accepted, see tests for `fake_clsig3` (fails without b010bc9). Chances to form such a quorum are close to 0 e.g. 1.29e-38 if you control 1500 MNs (see `Success probability of creating a malicious ChainLock` [here](https://github.com/dashpay/dips/blob/master/dip-0008.md#calculations) for more info), so it's not a critical issue by any means but I feel like it still makes sense to reduce any potential impact of such an event when possible.

Note: the issue is only for _future_ clsigs, the best clsig with a known block is not affected, no reorgs _below_ the already locked block are possible.

### CODE REVIEW

File: src/llmq/quorums_chainlocks.cpp
Comment (xdustinface) [inline src/llmq/quorums_chainlocks.cpp:144]: ^

**Review by xdustinface — CHANGES_REQUESTED** [review]
Good catch! 👍 See [suggestions-3978](https://github.com/xdustinface/dash/tree/suggestions-3978) for a few potential (trivial) tweaks in the test. In any case it looks like we need a rebase here after #3976 before an approval (if we merge that one first).

**Comment by UdjinM6** [comment]
Rebased

**Review by xdustinface — APPROVED** [review]
utACK

**Comment by PastaPastaPasta** [comment]
I'm not sure how I feel about this... Really, I'd say this is the best case for a malicious chainlock. If someone forms a malicious chainlock, the best outcome is the chain halting, because that prevents any attacker from actually making any money or anything off of the attack.

**Comment by UdjinM6** [comment]
@PastaPastaPasta I don't understand... can you elaborate pls?

**Pull request closed without merging**
